### PR TITLE
chore: fix push opened logic

### PR DESCRIPTION
**Pull request by alanjcharles**
- adds push delivered to notification service
- changes push open logic to remove `fromLaunch` bool

### CODE REVIEW

File: Example/BasicExample/TwiliEngageExampleNotificationService/NotificationService.swift [hunk 1]
```diff
@@ -6,6 +6,13 @@
 //
 
 import UserNotifications
+import Segment
+
+extension Analytics {
+    static var main = Analytics(configuration: Configuration(writeKey: "<WRITE_KEY>")
```
Comment (wenxi-zeng) [inline Example/BasicExample/TwiliEngageExampleNotificationService/NotificationService.swift:12]: may need a way to get the writekey from the analytics that user created?

**Review by wenxi-zeng — APPROVED** [review]

File: Example/BasicExample/TwiliEngageExampleNotificationService/NotificationService.swift [hunk 2]
```diff
@@ -39,7 +46,12 @@ class NotificationService: UNNotificationServiceExtension {
                 bestAttemptContent.attachments = [ mediaAttachment ]
             }
             
-            
+            var  notification = [String : Any]()
+            for (key, value) in bestAttemptContent.userInfo {
+                notification[key as! String] = value
+            }
+
+            Analytics.main.track(name: "Push Delivered", properties: notification)
```
Comment (alfrimpong) [inline Example/BasicExample/TwiliEngageExampleNotificationService/NotificationService.swift:54]: Im a bit concerned that we have to rely on the user to emit Push Delivered. Could the SDK do this?

File: Example/BasicExample/TwiliEngageExampleNotificationService/NotificationService.swift (hunk 1 above)
Comment (alanjcharles) [reply Example/BasicExample/TwiliEngageExampleNotificationService/NotificationService.swift:12]: they can do that- there's no way to access it from the extension bc `analytics` is always `nil` sooo they can add a constants folder or whatever

File: Example/BasicExample/TwiliEngageExampleNotificationService/NotificationService.swift (hunk 2 above)
Comment (alanjcharles) [reply Example/BasicExample/TwiliEngageExampleNotificationService/NotificationService.swift:54]: no- this is how one-trust and other competitors handle it. The Service Extension is the only place we can access a push payload without the app coming into the foreground. It's not pretty but Apple really limits us here.

**Pull request merged**